src/mapper/pkg/metricexporter/otel_edge_metric.go

```diff
@@ -0,0 +1,93 @@
+package metricexporter
+
+import (
+	"context"
+	"time"
+
+	"github.com/otterize/network-mapper/src/mapper/pkg/config"
+	sharedconfig "github.com/otterize/network-mapper/src/shared/config"
+	"github.com/spf13/viper"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/exporters/otlp/otlpmetric/otlpmetricgrpc"
+	"go.opentelemetry.io/otel/exporters/stdout/stdoutmetric"
+	"go.opentelemetry.io/otel/metric"
+	sdk "go.opentelemetry.io/otel/sdk/metric"
+	"go.opentelemetry.io/otel/sdk/resource"
+	semconv "go.opentelemetry.io/otel/semconv/v1.21.0"
+)
+
+type OtelEdgeMetric struct {
+	meterProvider *sdk.MeterProvider
+	counter       metric.Int64Counter
+}
+
+func newResource() (*resource.Resource, error) {
+	return resource.Merge(resource.Default(),
+		resource.NewWithAttributes(semconv.SchemaURL,
+			semconv.OTelLibraryName("otterize/network-mapper"),
+		))
+}
+
+const ClientAttributeName = "client"
+const ServerAttributeName = "server"
+
+func newMeterProvider(ctx context.Context, res *resource.Resource) (*sdk.MeterProvider, error) {
+	// SDK automatically configured via environment variables:
```

Comment:
I like this approach, but it does mean that the config is now in two places: the application config via viper and the OTEL config via its environment variables.

As we prove this out, there is probably a good reason to expose some of that config via the viper.